### ISSUE TITLE
Keyboard navigation doesn't work properly on registration form

### ISSUE THREAD

**Issue by attaxia** [AmpersandTarski/RAP#98]
#### De situatie
Keyboard navigation (handling of TAB key) does not work properly on the sign up form.

#### Wat er gebeurde er en wat had je verwacht?
It is not possible to TAB to the password field.

#### Stappen om dit te reproduceren
1. Go to registration form by clicking on the `Registration`-button
2. Enter a username
3. Press TAB to go to password field
4. Input focus is returned to username field

#### Screenshot / Video

[Screen Recording 2020-12-01 at 13.59.03.zip](https://github.com/AmpersandTarski/RAP/files/5622697/Screen.Recording.2020-12-01.at.13.59.03.zip)

**Comment by hanjoosten** [comment]
This is indeed very unfortunate, and it is known behavior.
You might also have noticed some other user unfriendly behavior on other screens. e. g. The screen where you can edit your script. You often have to explicitly click out of the current box to let the application know that it is time to update its state.

This lack of user friendlynes is a price to pay for having a generated application. We do our best to reduce this cost when possible.

**Comment by stefjoosten** [comment]
## A slightly more technical explanation
(for the interested reader)
RAP needs a click to trigger the process of checking rules. After you type the user name, the rule checker needs to be triggered. That happens at the next click. Whenever the rule checker kicks in, you will see that the screen turns green for a few seconds, indicating that no violations were found. Or you will see that an error message pops up, in case of which the window turns red.

You would like to see that tab has a similar effect (i.e. trigger the rule checker) and brings you to the next open field at the same time. You are quite right in wanting this, because the current behaviour is pretty annoying.This is something that we cannot fix easily, because it is in the runtime behavior of generated code. But we have it on the bucket list, so in due time it will come.... Just don't know when.

**Comment by attaxia** [comment]
It is not only annoying, and I get where it comes from, but it also causes severe issues for people with disabilities depending on keyboard navigation or accessibility software.

**Comment by stefjoosten** [comment]
I'm very sorry for that. I am aware of the ISO-standard for inclusive design. It is for sure that RAP does not comply. Alas, we cannot fix it in the short term.
